common/cmake/devices.cmake

```diff
@@ -279,37 +279,17 @@ function(DEFINE_DEVICE_TYPE)
   set(DEVICE_UNIQUE_PACK_FILE arch.unique_pack.xml)
   set(DEVICE_MERGED_LINT_FILE arch.merged.lint.html)
 
-  set(MERGE_XML_XSL ${symbiflow-arch-defs_SOURCE_DIR}/common/xml/xmlsort.xsl)
-
-  get_file_location(MERGE_XML_INPUT ${DEFINE_DEVICE_TYPE_ARCH_XML})
-  append_file_dependency(DEPS ${DEFINE_DEVICE_TYPE_ARCH_XML})
-
   set(MERGE_XML_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${DEVICE_MERGED_FILE})
   set(UNIQUE_PACK_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${DEVICE_UNIQUE_PACK_FILE})
   set(MERGE_XMLLINT_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${DEVICE_MERGED_LINT_FILE})
 
-  get_target_property_required(XSLTPROC env XSLTPROC)
-  get_target_property(XSLTPROC_TARGET env XSLTPROC_TARGET)
-  set(ARCH_SCHEMA ${symbiflow-arch-defs_SOURCE_DIR}/common/xml/fpga_architecture.xsd)
-
-  add_custom_command(
-    OUTPUT ${MERGE_XML_OUTPUT}
-    DEPENDS
-      ${MERGE_XML_XSL}
-      ${DEPS}
-      ${XSLTPROC} ${XSLTPROC_TARGET}
-    COMMAND
-      ${CMAKE_COMMAND} -E make_directory
-      ${CMAKE_CURRENT_BINARY_DIR}/${OUT_DEVICE_DIR}
-    COMMAND
-      ${XSLTPROC}
-      --nomkdir
-      --nonet
-      --xinclude
-      --output ${MERGE_XML_OUTPUT} ${MERGE_XML_XSL} ${MERGE_XML_INPUT}
+  xml_sort(
+    NAME ${DEFINE_DEVICE_TYPE_ARCH}_${DEFINE_DEVICE_TYPE_DEVICE_TYPE}_arch_merged
+    FILE ${DEFINE_DEVICE_TYPE_ARCH_XML}
```

Comment:
Maybe "INPUT" over "FILE"

Reply:
Doesn't this need to match the definition in `common/cmake/xml.cmake`?

Reply:
> Doesn't this need to match the definition in `common/cmake/xml.cmake`?

It does, I thought xml_sort was introduced in this PR and I was commenting about FILE being a so-so argument name.

Reply:
It already existed, I just made it used by this function.

Would either of these be better names?
 * `xml_canonicalize_merge` 
 * `xml_prepare_for_vpr`

Reply:
FYI - The tool originally just sorted the xml file but then it gained more "features"....

Reply:
I think ``xml_canonicalize_merge`` explains what it does well.  ``xml_prepare_for_vpr`` says why we do it.  I think I like ``xml_canonicalize_merge`` better.

Reply:
Done.